src/compiler/scala/tools/nsc/transform/ExplicitOuter.scala

```diff
@@ -315,17 +315,17 @@ abstract class ExplicitOuter extends InfoTransform
     * (1) An class which is not an interface and is not static gets an outer accessor (@see outerDefs).
     * (1a) A class which is not a trait gets an outer field.
     *
-    * (4) A constructor of a non-trait inner class gets an outer parameter.
+    * (4) A constructor or static method of a non-trait inner class gets an outer parameter.
```

Comment:
Minimally, all it needs to admit right now is `sym.isStaticMember && sym.isDefaultGetter`).

I'm considering future use cases for static methods, added before `explicitouter`, where an outer parameter would be unnecessary and unwanted. For instance, if we had a `@static` annotation that let people write real static methods for Java interop.

I suppose we can just change this code if/when the other use case emerges.

Reply:
I guess we don't have a mode where an outer parameter is only added when it's actually used. So this would be disallowed?
```
class C {
  var x = 0
  class D {
    @static def f = x
  }
}
```
Anyway, I'm fine narrowing the condition to constructor default getters for now.